### PR TITLE
ST-855: added building complex types

### PR DESCRIPTION
**Pull request by sharmad-v1**
### Change description ###

Enter a description.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SPTRIBS-

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/sptribs/document/dto/CicBundleDocumentDTO.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Builder
+@Slf4j
+public class CicBundleDocumentDTO {
+
+    private String name;
+    private String description;
+    private int sortIndex;
+    private CICDocument sourceDocument;
+
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/document/dto/CicBundleDocumentDTO.java:17]: Can use Lombok @AllArgsConstructor and @NoArgsConstructor here?

**Review by osheaf01 — CHANGES_REQUESTED** [review]
Can you look into using Lombok for constructors, getters and setters, please?

File: src/main/java/uk/gov/hmcts/sptribs/document/dto/CicBundleDocumentDTO.java (hunk 1 above)
Comment (sharmad-v1) [reply src/main/java/uk/gov/hmcts/sptribs/document/dto/CicBundleDocumentDTO.java]: Review comment fixed.

File: src/main/java/uk/gov/hmcts/sptribs/document/dto/CicBundleDocumentDTO.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        log.info("CicBundleDocumentDTO no args constructor");
+    }
+
+    public CicBundleDocumentDTO(String name, String description, int sortIndex, CICDocument sourceDocument) {
+        this.name = name;
+        this.description = description;
+        this.sortIndex = sortIndex;
+        this.sourceDocument = sourceDocument;
+    }
+
```
[the remark this answers is not in the document]
Comment (sharmad-v1) [reply src/main/java/uk/gov/hmcts/sptribs/document/dto/CicBundleDocumentDTO.java]: Review comment fixed.

**Comment by sharmad-v1** [comment]
> Can you look into using Lombok for constructors, getters and setters, please?

Changes done.

**Comment by sharmad-v1** [comment]
> 

Added lombok setter, getter, No Args and AllArgs contructor.

File: src/main/java/uk/gov/hmcts/sptribs/document/dto/CicBundleDocumentDTO.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sharmad-v1) [reply src/main/java/uk/gov/hmcts/sptribs/document/dto/CicBundleDocumentDTO.java]: Added Lombok.

**Review by osheaf01 — APPROVED** [review]
LGTM

**Pull request closed without merging**